### PR TITLE
#13: Config Tweak & Integration pipeline

### PR DESCRIPTION
**Pull request by NathHorrigan**
In this PR I have:

- Written an example [Jest](https://jestjs.io/) and [RNTL](https://github.com/callstack/react-native-testing-library) test.
- Updated Babel configuration to automatically `import React` if it detects JSX in the file.
- Updated React to `16.14.0`
- Tweaked Prettier configuration
- Added Github Actions to run linting and unit tests, These have been tested locally using [act](https://github.com/nektos/act)

### CODE REVIEW

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Code Checker
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
Comment (mixxorz) [inline .github/workflows/node.yml]: This should be `main`

File: .github/workflows/node.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    # Install dependencies and lint the typescript using prettier
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.x'
```
Comment (mixxorz) [inline .github/workflows/node.yml]: Suggestion: Use Node 12.x as this is the LTS version.

We should also probably specify this in the `engines` section in `package.json`.

File: package.json [hunk 3]
```diff
@@ -7,11 +7,12 @@
     "@react-navigation/stack": "^5.9.3",
     "@reduxjs/toolkit": "^1.4.0",
     "@shopify/restyle": "^1.3.0",
+    "@types/jest": "^26.0.15",
     "expo": "~39.0.0",
     "expo-splash-screen": "~0.6.2",
     "expo-status-bar": "~1.0.2",
     "expo-updates": "~0.3.2",
-    "react": "16.13.1",
+    "react": "^16.14.0",
```
Comment (mixxorz) [inline package.json:18]: Q: Are we sure we can keep the `react` version decoupled from the `react-native` version?

File: src/components/__tests__/__snapshots__/Text.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+<Text
+  style={
+    Array [
+      Object {
+        "display": "flex",
+      },
+    ]
+  }
+/>
+`;
```
Comment (mixxorz) [inline src/components/__tests__/__snapshots__/Text.test.tsx.snap:13]: Interesting! Hopefully not to tedious to keep up to date.

**Review by mixxorz — COMMENTED** [review]
Just a few minor issues. Overall looks pretty good.

Question: Why didn't the actions run for this PR? Is this something we need to enable in the settings?

File: .github/workflows/node.yml (hunk 1 above)
Comment (NathHorrigan) [reply .github/workflows/node.yml]: Thank You!

File: src/components/__tests__/__snapshots__/Text.test.tsx.snap (hunk 4 above)
Comment (NathHorrigan) [reply src/components/__tests__/__snapshots__/Text.test.tsx.snap:13]: It's automated ;)

File: .github/workflows/node.yml (hunk 2 above)
Comment (NathHorrigan) [reply .github/workflows/node.yml]: Done!

File: package.json (hunk 3 above)
Comment (NathHorrigan) [reply package.json:18]: Yes, it should be fine. `16.14.0` Only added the JSX package and a couple bug fixes :)

**Review by mixxorz — APPROVED** [review]
Looks good. Thanks Nathan!

**Pull request merged**